### DISCUSSION TITLE
How to create background service in WindowsAppSdk ?

### DISCUSSION THREAD

**Discussion by AmolGuldagad** [microsoft/WindowsAppSDK#4162] [Q&A]
How to create background service in WindowsAppSdk ? 
We want to create background service in WindowsAppSdk which will be running all the time with or without running main application.
Need combination of main application(has UI) & background service.
Even though main application is not running, we want background service to run always and perform some actions as require.
On windows startup also - want to start background service

**Comment by DarranRowe** [comment]
I have two questions.

1) What do you mean by "background service". It implies a certain type of application to me, and what it implies has nothing to do with the Windows App SDK. I just want to be sure that you are thinking of the same thing that I am here.
2) Are you sure that you need to use the Windows App SDK? The Windows App SDK is a library, not a framework. So unless you need to features of the Windows App SDK, then you don't need to use anything but the existing Windows functionality.

This does raise the question, how much of the Windows App SDK works properly when used in a Windows service application.

**Reply by AmolGuldagad** [reply]
Hi @DarranRowe 
We have created winUI3 application using "Blank App, Packaged (WinUI 3 in Desktop)"

Goal - we want to create WinUI3 application for UI and in background, there should be one process always running which handles certain events and perform some actions.
MainApp (create WinUI3 project with output type = windows Application) and 
Background service (created class library and converted into output type = windows Application)) 

**Consider Flow -** 

- in main Application, we can register background task
- Background service/process(independent exe) is in always running state 
- main application is not running 
- on certain xyz event - background service want to notify background task of main Application and
- in Background task(event trigger) of mainApp - load required dll using reflection of Module1 or Module 2 and execute method of that dll.

How to create this scenario? 
![image](https://github.com/microsoft/WindowsAppSDK/assets/40460016/e1bdebef-a8ad-4212-a5be-e449d1c5a3cc)
![image](https://github.com/microsoft/WindowsAppSDK/assets/40460016/1288fd14-9b26-4ba7-b7d6-5e3aa69076f3)
